app/services/spree_shopify_importer/data_parsers/stock_items/base_data.rb

```diff
@@ -0,0 +1,37 @@
+module SpreeShopifyImporter
+  module DataParsers
+    module StockItems
+      class BaseData
+        def initialize(spree_stock_location, inventory_level)
+          @spree_stock_location = spree_stock_location
+          @inventory_level = inventory_level
+        end
+
+        def variant_data_feed
+          SpreeShopifyImporter::DataFeed
+            .where(shopify_object_type: 'ShopifyAPI::Variant')
+            .where('data_feed like ?', "%inventory_item_id\":#{@inventory_level.inventory_item_id}%").first
+        end
+
+        def variant
+          Spree::Variant.find(variant_data_feed.spree_object_id)
+        end
+
+        def stock_item_attributes
+          {
+            stock_location_id: @spree_stock_location.id,
+            variant_id: variant.id
+          }
+        end
+
+        def backorderable?
+          @backorderable ||= JSON.parse(variant_data_feed.data_feed)['inventory_policy'].eql?('continue')
```

Comment:
no, I'm not, when do I need? it is the practice in this project, it is bad?

Reply:
no, it's not bad, but you should use memoization **only** in case when you call method twice or more within life cycle of the object, but by searching I found that this method is used only once, so memoization here doesn't make sense, if I'm right please remove it